### ISSUE TITLE
False Negatives in DMI_CONSTANT_DB_PASSWORD and DMI_EMPTY_DB_PASSWORD Rules

### ISSUE THREAD

**Issue by tacu22** [spotbugs/spotbugs#3990]
**Describe the bug**
The current implementations of `DMI_CONSTANT_DB_PASSWORD` and `DMI_EMPTY_DB_PASSWORD` have some false negatives. They strictly match the signature `DriverManager.getConnection(String, String, String)`, failing to detect hardcoded credentials when passed via Properties objects, embedded directly in the JDBC URL, or stored in local/static variables before invocation. 

Affected Rules
- DMI_CONSTANT_DB_PASSWORD
- DMI_EMPTY_DB_PASSWORD

**To Reproduce**
```java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class IssueDBCredentialsDetection {
    
    private static final String STATIC_PASS = "static_cred"; // Fails: Constant Field Propagation
    
    public void connectTests() throws SQLException {
        
        // 1. Standard Detection (Works)
        DriverManager.getConnection("jdbc:mysql://localhost", "user", "direct_hardcoded");
        
        // 2. Properties Overload (Fail)
        Properties props = new Properties();
        props.setProperty("password", "prop_cred"); // Not scanned
        DriverManager.getConnection("jdbc:mysql://localhost", props);
        
        // 3. URL Embedded Credentials (Fail)
        DriverManager.getConnection("jdbc:mysql://localhost/db?user=admin&password=url_sec");
        
        // 4. Local Variable Propagation (Fail)
        String dynamicPass = STATIC_PASS;
        DriverManager.getConnection("jdbc:mysql://localhost", "admin", dynamicPass);
        
        // 5. Empty Password in Variable (Fail)
        String emptyPass = ""; 
        DriverManager.getConnection("jdbc:mysql://localhost", "root", emptyPass);
    }
}
```

**Expected behavior**

**Environment**
- Java Compiler: [javac 21.0.10]
- Java version: [JDK 21]
- SpotBugs Version: [4.9.8]
- How do you use SpotBugs: [CLI]

**Additional context**
While I have found that there are two issues (https://github.com/spotbugs/spotbugs/issues/3633, https://github.com/spotbugs/spotbugs/issues/3529) that seem similar to cases 4 and 5, I have retained them here to ensure a comprehensive scope for fully refining this rule.
